src/core/ddsi/include/dds/ddsi/q_rtps.h

```diff
@@ -74,6 +74,8 @@ typedef int64_t seqno_t;
 #define NN_ENTITYID_KIND_WRITER_NO_KEY 0x03
 #define NN_ENTITYID_KIND_READER_NO_KEY 0x04
 #define NN_ENTITYID_KIND_READER_WITH_KEY 0x07
+#define NN_ENTITYID_KIND_CYCLONE_TOPIC_BUILTIN 0x0c // Vendor specific
```

Comment:
When allocating a new entity ID (in `ddsi_new_topic`), the flag `NN_ENTITYID_SOURCE_VENDOR` is added so that the topic entities are marked as vendor specific. I think that the two kinds we add here should fit in the kind mask (0x3f), and I also think the kind 'topic' should actually have been defined in the RTPS specification. The reason for defining two topic kinds here (which is far from ideal), is because built-in and vendor-specific flag values cannot be combined (see RTPS 2.3 spec section 9.3.1.2)